source/dofs/dof_accessor_set.cc

```diff
@@ -41,91 +41,235 @@
 
 DEAL_II_NAMESPACE_OPEN
 
+template <typename Number>
+DeclException2(ExcNonMatchingElementsSetDofValuesByInterpolation,
+               Number,
+               Number,
+               << "Called set_dof_values_by_interpolation(), but"
+               << " the element to be set, value " << std::setprecision(16)
+               << arg1 << ", does not match with the value "
+               << std::setprecision(16) << arg2 << " already set before.");
+
+namespace internal
+{
+  template <typename Number>
+  typename std::enable_if<
+    !std::is_unsigned<Number>::value,
+    typename numbers::NumberTraits<Number>::real_type>::type
```

Comment:
With C++14, you can use this instead:
```suggestion
  std::enable_if_t<
    !std::is_unsigned<Number>::value,
    typename numbers::NumberTraits<Number>::real_type>
```